Fix several issues related to cross compilation

Jake-Shadle opened the pull request:
This fixes various compiler errors/warnings that occur when building for Windows using clang-cl as the compiler.

Most of the changes boil down to a conflation of Windows and MSVC being the same thing, though there were a few warnings produced by clang for code that was only ever compiled with MSVC before, 

Also, the `WinSock2.h` was included with the wrong casing.

This includes 2 other compilation fixes for #190, #183

repi committed:
Fix compile warning/error on Xcode 11.

Disabled the "atomic-implicit-seq-cst" warning that seems to have been enabled by default going from Xcode 10 to 11 (and then new clang version).

This is tested to also compile on Xcode 10.3, which also had the warning but wasn't enabled by default.

Fixes #183 and https://github.com/EmbarkStudios/physx-rs/issues/23.

h3r2tic committed:
Fix obsolete typeinfo.h include on Windows

Jake-Shadle committed:
Use correct casing for WinSock2.h

Jake-Shadle committed:
Remove unneeded PX_LINUX guard

Jake-Shadle committed:
Fix conflation of target and compiler

Jake-Shadle committed:
Fix conflation of target and compiler

Jake-Shadle committed:
Fix conflation of target and compiler

Jake-Shadle committed:
Fix for cross compiling to windows with clang

Jake-Shadle committed:
Fix for clang warning

Jake-Shadle committed:
Fix cross compilation for windows with clang

Jake-Shadle committed:
Fix undef warning

Jake-Shadle committed:
Fix more clang/windows issues

Jake-Shadle committed:
Fix more clang warnings

Jake-Shadle committed:
Fix more clang warnings

Jake-Shadle committed:
Merge pull request #1 from EmbarkStudios/4.1-embark

Cross compilation fixes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/NVIDIAGameWorks/PhysX?pullRequest=269) <br/>All committers have signed the CLA.

repi committed:
Name threads on Windows with SetThreadDescription
This enable profilers and others tools to see the thread names, not just the debugger.
Still use the RaiseException method also because it works on all Windows versions

Jake-Shadle committed:
Fix misleading indentationhttps://clang.llvm.org/docs/DiagnosticsReference.html#wmisleading-indentation

Jake-Shadle committed:
Add missing const

Jake-Shadle committed:
Disable warning about potential divide by 0

Jake-Shadle committed:
Apparently that doesn't work
